manual_benchmarks/test_datatypes_crunch.ipynb

```diff
@@ -0,0 +1,283 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "test_datatypes_crunch.ipynb",
+      "provenance": []
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CqyExbrNYKcv"
+      },
+      "source": [
+        "### Import stuff"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "wF4Ga_unYJ8F"
+      },
+      "source": [
+        "!pip install radis\n",
+        "\n",
+        "from radis.db.classes import get_molecule_identifier\n",
+        "from radis.levels.partfunc import PartFuncHAPI\n",
+        "\n",
+        "from radis.io.hitemp import fetch_hitemp\n",
+        "from radis.db.classes import get_molecule\n",
+        "from radis.phys.constants import hc_k\n",
+        "\n",
+        "import numpy as np\n",
+        "from numpy import exp, pi"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jhglpoCFXrpp"
+      },
+      "source": [
+        "### Function to reduce the memory usage of pandas dataframe"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "CryqdLGaXdFO"
+      },
+      "source": [
+        "def reduce_mem_usage(props):\n",
+        "    start_mem_usg = props.memory_usage().sum() / 1024**2 \n",
+        "    print(\"Memory usage of properties dataframe is :\",start_mem_usg,\" MB\")\n",
+        "    NAlist = [] # Keeps track of columns that have missing values filled in. \n",
+        "    for col in props.columns:\n",
+        "        if props[col].dtype != object:  # Exclude strings\n",
+        "            \n",
+        "            # Print current column type\n",
+        "            print(\"******************************\")\n",
+        "            print(\"Column: \",col)\n",
+        "            print(\"dtype before: \",props[col].dtype)\n",
+        "            \n",
+        "            # make variables for Int, max and min\n",
+        "            IsInt = False\n",
+        "            mx = props[col].max()\n",
+        "            mn = props[col].min()\n",
+        "            \n",
+        "            # Integer does not support NA, therefore, NA needs to be filled\n",
+        "            if not np.isfinite(props[col]).all(): \n",
+        "                NAlist.append(col)\n",
+        "                props[col].fillna(-1,inplace=True)  \n",
+        "                   \n",
+        "            # test if column can be converted to an integer\n",
+        "            asint = props[col].fillna(0).astype(np.int64)\n",
+        "            result = (props[col] - asint)\n",
+        "            result = result.sum()\n",
```

Comment:
When doing this, always take the absolute value before summation, otherwise you may be canceling positive and negative errors with each other.

You may also want to look and see if np.isclose() is useful here.